build/dist

```diff
@@ -279,6 +295,12 @@ if [[ "$SPARK_PROVIDED" != "true" ]]; then
   cp -r "$SPARK_BUILTIN" "$DISTDIR/externals/"
 fi
 
+if [[ "$HIVE_PROVIDED" != "true" ]]; then
```

Comment:
not related to this PR only, if we build/package without hive distribution with maven, the below will fail

Reply:
I forgot, this PR needs to be merged in https://github.com/apache/incubator-kyuubi/issues/2024 to work properly.

Reply:
>I forgot, this PR needs to be merged in https://github.com/apache/incubator-kyuubi/issues/2024 to work properly.

yes, you are right. 

> not related to this PR only, if we build/package without hive distribution with maven, the below will fail

This is another issue we may face if we `build/dist ` with download releases skipped.

Reply:
I probably know what you mean.
If you mean to use maven -D parameter to skip the release download of hive or other engines, an error will be reported here.
```
build/dist --tgz --spark-provided --flink-provided  -Pspark-3.2 -Dhive.archive.download.skip=true
```

Reply:
Maybe do this after `#Parse arguments`.

```bash
mvn help:evaluate -Dexpression=hive.archive.download.skip -Dhive.archive.download.skip=false  | grep -v INFO | grep -v WARNING
```

Do I need to make changes in this PR? or new open issue to do this?

Reply:
Let‘s fix this in another ticket